temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -613,6 +614,43 @@ public void completeWorkflowTask(
         request.hasStickyAttributes() ? request.getStickyAttributes() : null);
   }
 
+  @Override
+  public void applyOnConflictOptions(@Nonnull StartWorkflowExecutionRequest request) {
```

Comment:
Note that we added validation that attachCompletionCallbacks cannot be true if attachRequestId is false in the server, you should validate that in the test server and reject such requests with an invalid argument error.